lib/git.zsh

```diff
@@ -12,11 +12,53 @@ function __git_prompt_git() {
 # Outputs current branch info in prompt format
 function git_prompt_info() {
   local ref
-  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" != "1" ]]; then
+  local branches
+  local info
+  local deatchedtag
+
+  # If oh-my-zsh.hide-status equals to '1', get out.
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.hide-status 2>/dev/null)" == "1" ]]; then
+    return 0
+  fi
+
+  # If we are on a folder not tracked by git, get out.
+  if [[ "$(__git_prompt_git rev-parse --git-dir 2> /dev/null)" == "" ]]; then
+    return 0
+  fi
+
+  # If oh-my-zsh.classic-info is set, use classic info output of branch only
+  if [[ "$(__git_prompt_git config --get oh-my-zsh.classic-info 2>/dev/null)" == "1" ]]; then
     ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
     ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null) || return 0
     echo "$ZSH_THEME_GIT_PROMPT_PREFIX${ref#refs/heads/}$(parse_git_dirty)$ZSH_THEME_GIT_PROMPT_SUFFIX"
+    return 0
+  fi
+
+  branches=$(__git_prompt_git branch --list -a 2> /dev/null)
+  if [[ "${branches}" == "" && "$?" == "0" ]]; then
+      # Look for initialized repos that have zero commits
+      ref=$(__git_prompt_git symbolic-ref HEAD 2> /dev/null) || \
+      ref=$(__git_prompt_git rev-parse --short HEAD 2> /dev/null)
+      info="<initialized>/${ref#refs/heads/}"
```

Comment:
I dislike that this adds `<initialized>` to the prompt. This is unnecessary and verbose. Just print the branch.

Reply:
It will display `<initialized>` only when in a repository without any commit. Literally, just got initialized. It helps differentiate against other repos that already have something in.

It will display `<no-remote>` only when in a repository *already* initialized (has at least one commit), but no remote.

In my experience with this enhanced git prompt, I have found these quite useful to help differentiate between multiple local folders/copies of the same project.

Reply:
Why would you need to say it's initialized? You already know it is, you just created the repo (?) Again, it's unnecessary and verbose (prompt line space is sacred, this adds 13 characters.